plugins/modules/system/disk_info.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/python
+#
+# Copyright 2021 VMware, Inc
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: disk_info
+version_added: "2.5.0"
+short_description: Retrieves disk info in the Linux/Unix systems
+description:
+    - This module retrieves disk info like free space, used percentage, mount location, and so on.
+requirements:
+  - psutil (Python module)
+options:
+    name:
+        description:
+            - This is the disk name to query, if this parameter is not passed, it queries all the disk information.
+        default: all
+        type: str
+    filter:
+        description:
+            - This is the parameter used to filter only the specific information from the output like freespace, mountpoint, fstype, and so on.
+        type: list
+        elements: str
+author:
+    - Saranya Sridharan (@saranyasridharan)
+    - Swetha M (@swetm)
+    - Nagendra Kini (@NagendraKini)
+    - Anup Kumar (@anuppoojari)
+    - Mehmoodkha Pathan (@Mehmoodpathan)
+'''
+
+EXAMPLES = '''
+- name: Get disk info
+  community.general.disk_info:
+
+- name: Get Disk info for Particular disk info
+  community.general.disk_info:
+    name: diskname
+
+- name: Filter the output parameters for specific disk
+  community.general.disk_info:
+    name: diskname
+    filter:
+      - mountpoint
+      - freespace
+      - totalsize
+
+- name: Filter the output parameters for all disk
+  community.general.disk_info:
+    filter:
+      - fstype
+      - usedspace
+      - capacity_percentage
+'''
+
+
+RETURN = '''
+ansible_facts:
+    description: disk information only filtered parameter values if filter option is given.
+    returned: dictionary of disks informations
+    type: dict
+    sample: {
+        '/dev/sda': {
+             'capacity_percentage': '2%'
```

Comment:
1. This is not correct anymore.
2. There are a lot more fields in here. You should at least mention them all in the only example you have.